### ISSUE TITLE
Bot comment on build failure

### ISSUE THREAD

**Issue by mmarchini** [nodejs/github-bot#251]
(not when the job fails to start, but when the tests or build fails, so it wouldn't revert https://github.com/nodejs/github-bot/issues/238)

`node-core-utils` has this great tool `ncu-ci pr`, which can be used to get a list of failing tests. We could use the GitHub Bot to comment the result of `ncu-ci pr` back on the Pull Request when a build fails, so it's easier for folks to see which errors happened. Or the bot could edit it's own comment, to avoid notifications, but personally I would like to get notified (I understand folks might be opposed to it though).

**Comment by phillipj** [comment]
To ensure I understand you correctly and having in mind I haven't used `ncu-ci` before; in addition to the inline PR statuses shown below, the Github Bot would also post a comment providing more details about the actual test(s) that failed? Meaning not just the fact that a job failed like the PR statuses says?

![Screenshot 2020-03-11 at 09 14 05](https://user-images.githubusercontent.com/1231635/76396079-d6408780-6378-11ea-8498-aeb40095d4e8.png)

**Comment by mmarchini** [comment]
Exactly. It would look like this: https://github.com/nodejs/node/pull/32116#issuecomment-597683266

**Comment by mmarchini** [comment]
I just thought of two other huge benefits of this:

1. We'll keep a history of CI failures (since Jenkins deletes old CI runs)
2. When searching for a flaky test, every PR where the test failed will show on the results (which will make it easier to see if the test is flaky than navigating Jenkins to find each occurrence of that test failing).

**Comment by phillipj** [comment]
Well said!

I definitively see the value of trying what you're proposing. Learning from well intentioned changes we've introduced before, that ended up getting reverted later primarily because we didn't reach out to core collaborators beforehand, I'm wondering if we could do some kind of collaborator vote or similar before jumping onto implementing this?

**Comment by mmarchini** [comment]
That might be a good idea. I also think the "edit comment" idea would be more likely to be well received. I'll start a poll, thank you for the suggestion.
